okhttp-tests/src/test/java/okhttp3/HeadersTest.java

```diff
@@ -56,6 +57,23 @@
     assertEquals("HTTP/1.1", headers.value(0));
   }
 
+  @Test public void readNameValueBlockDropsHeadersUpTo100Continue() throws IOException {
```

Comment:
Thought this test confirms that the implementation of readHttp2HeadersList does what it should, that method is a deep implementation detail that’s quite likely to get refactored in future. A more useful test would exercise the “user’s view” of these headers . . . that an actual HTTP/2 response gets the right headers.

Reply:
I think we need the implementation around `httpSink` to write this test in a more instrumental fashion.